### ISSUE TITLE
Build job generated four times automatically.

### ISSUE THREAD

**Issue by shiningmosquito** [drone/drone#2094]
Hi there,

My Drone integrated with a Gitlab private repository.
When a commit sent to my repository, the build job generated four times automatically.

Drone version: 0.7.3

My pipeline flow as below:
1. Clone Gitlab source code.
2. Install node module (image: kkarczmarczyk/node-yarn:6.7-wheezy).
3. Build Docker image and push to AWS ECR (image: plugins/ecr).
4. Deploy to AWS ECS (image: peloton/drone-ecs).

The screenshot of my Drone build job list:
![image](https://user-images.githubusercontent.com/8151846/27670600-8c795214-5c53-11e7-90f1-8652a8c97240.png)

Have someone ever had the same situation?

**Comment by amq** [comment]
I have the same problem. You could say that it is GitLab's fault, but:
- other CI systems have found a way to work with GitLab
- Drone claims to have a full GitLab support, without any word of warning in docs
- why allow *remotely* starting the build multiple times for the same commit, before even a single build has finished?
- why close an issue without a word of explanation?

**Comment by bradrydzewski** [comment]
>other CI systems have found a way to work with GitLab

this impacts other individuals and other CI solutions such as appveyor. See https://gitlab.com/gitlab-com/support-forum/issues/350

> why close an issue without a word of explanation?

We triaged the problem in the chatroom and found that Gitlab was submitting multiple hooks erroneously. The issue was closed since this was not a drone issue.

> Drone claims to have a full GitLab support, without any word of warning in docs

I believe when triaging we found this was a Gitlab problem related to Nginx proxies not terminating connections properly causing GitLab to resend hooks. So the assertion that Drone does not fully support GitLab is not necessarily true, and may simply be product of your specific configuration.

> why allow remotely starting the build multiple times for the same commit, before even a single build has finished?

This issue only impacts _some_ GitLab configurations due to a known GitLab bug, likely only experience with a reverse proxy, and perhaps more specifically, Nginx. I see no reason for Drone to workaround this problem when the GitLab team is adequately resourced to solve this sort of bug when compared to the Drone team which is not venture back and has no full time employees.

So tldr; Go complain somewhere else.
